packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/Converters.kt

```diff
@@ -70,22 +76,47 @@ internal interface PublicConverter<T, S> {
  * This corresponds to step 2. of the overall conversion described in the top of this file.
  */
 internal interface StorageTypeConverter<T> {
-    public fun fromRealmValue(realmValue: RealmValue): T? = realmValueToAny(realmValue) as T?
-    public fun toRealmValue(value: T?): RealmValue = anyToRealmValue(value)
+    fun fromRealmValue(realmValue: RealmValue?): T?
+    fun MemAllocator.toRealmValue(value: T?): RealmValue
 }
 // Top level methods to allow inlining from compiler plugin
-public inline fun realmValueToAny(realmValue: RealmValue): Any? = realmValue.value
-public inline fun anyToRealmValue(value: Any?): RealmValue = RealmValue(value)
+public inline fun realmValueToLong(transport: RealmValue?): Long? =
+    transport?.getLong()
+public inline fun realmValueToBoolean(transport: RealmValue?): Boolean? =
+    transport?.getBoolean()
+public inline fun realmValueToString(transport: RealmValue?): String? =
+    transport?.getString()
+public inline fun realmValueToByteArray(transport: RealmValue?): ByteArray? =
+    transport?.getByteArray()
+public inline fun realmValueToRealmInstant(transport: RealmValue?): RealmInstant? =
+    transport?.let { RealmInstantImpl(it.getTimestamp()) }
+public inline fun realmValueToFloat(transport: RealmValue?): Float? =
+    transport?.getFloat()
+public inline fun realmValueToDouble(transport: RealmValue?): Double? =
+    transport?.getDouble()
+public inline fun realmValueToObjectId(transport: RealmValue?): BsonObjectId? =
+    transport?.let { it.getObjectId() }
+public inline fun realmValueToRealmObjectId(transport: RealmValue?): ObjectId? =
+    transport?.let { ObjectIdImpl(it.getObjectId().toByteArray()) }
+public inline fun realmValueToRealmUUID(transport: RealmValue?): RealmUUID? =
+    transport?.let { RealmUUIDImpl(it.getUUIDWrapper()) }
```

Comment:
Again extracting wrapped types instead of just converting the transport types directly into the public types. Do we really need the wrappers?

Reply:
Removed them.